registration/views.py

```diff
@@ -231,6 +231,17 @@ def get(self, request, *args, **kwargs):
         clean_session(request.session)
         clean_next_of_kin_email_session(request.session)
         setup_external_source_session(request)
+        if self.request.GET.get('email') and self.request.GET.get('postal_code'):
+            email = self.request.GET['email']
+            postal_code = self.request.GET['postal_code']
+            # Email and postal codes are provided, go ahead and process them
+            try:
+                self.submit_email({'email': email, 'postal_code': postal_code})
```

Comment:
@GhalebKhaled Can you please try to initialize and submit the form as following?

```
form = StateLookupForm(**initial**={'email': email, 'postal_code': postal_code})
if form.is_valid():
> return self.form_valid(form)
else:
> return self.form_invalid(form)
```